content/realtime/metachannels.textile

```diff
@@ -115,17 +116,150 @@ metaChannel.subscribe('channel.region.inactive', (msg) => {
+When @channel.occupancy@ "events are enabled":https://knowledge.ably.com/how-can-i-enable-channel-lifecycles for your app, you can subscribe to channel occupancy events by attaching to the channel @[meta]channel.lifecycle@ using a realtime connection. However, please bear in mind that "realtime connections and channels have rate limits":https://knowledge.ably.com/do-you-have-any-connection-message-rate-or-other-limits-on-accounts based on your package size. If the rate of channel lifecycle events exceeds this, a realtime connection is not going to be a reliable way to consume these events as you may get disconnected or lifecycle events may be discarded.
 
 While the channel lifecycle metachannel can provide occupancy metrics, there may be a performance impact, since this involves funnelling occupancy updates for every channel in the app to a single metachannel. This approach does not scale well as the number of channels grows, and so has limited use cases. There are two recommended alternatives:
 
 * "Inband channel occupancy events":/realtime/inband-occupancy - for use cases where the entity that needs to determine occupancy is attached to the channel. 
 * Occupancy rules - for use cases where the entity needing to determine channel occupancy is not connected to the channel. A channel occupancy reactor rule allows you to obtain occupancy updates from a set of channels in a centralized way, that doesn't involve attaching to all of them. Please "get in touch":mailto:enterprise-sales@ably.io for further information.
 
-h3(#reactor). Reactor rules
+h2(#sampled-events). Sampled connection and request events
+
+Sampled connection and request events can be used to observe the usage of Ably services. Events are published to metachannels that contain metadata for a sample of connections and API requests. These events can be used to compile statistics on app usage, enabling you to perform arbitrary data processing and aggregation related to client population and client activity.
+
+Note that this is an Enterprise-only feature and must be enabled for each app you would like to sample. Please "contact us":https://ably.com/contact to enable this functionality.
+
+Events are published on the metachannels *@[meta]clientEvents:connections@* and *@[meta]clientEvents:apiRequests@*.
+
+The following code snippet is an example of subscribing to connection events on the @[meta]clientEvents:connections@ channel:
+
+```[javascript]
+const channel = ably.channels.get('[meta]clientEvents:connections');
+
+channel.subscribe('connection.opened', (msg) => {
+    console.log('connection opened: ', msg.data);
+})
+
+channel.subscribe('connection.closed', (msg) => {
+    console.log('connection closed: ', msg.data);
+})
+
+channel.subscribe('connection.failed', (msg) => {
+    console.log('connection failed: ', msg.data);
+})
+
+channel.subscribe('connection.refused', (msg) => {
+    console.log('connection refused: ', msg.data);
+})
```

Comment:
Showing the client subscribing to every message name there is one by one is a little bit of an antipattern -- if you want to subscribe to everything you can just do that. If you want to show both, maybe:
```js
// Subscribe to all event types
channel.subscribe((msg) => {
    console.log('Event type: ' + msg.name, msg.data);
});

// Subscribe to one particular event type
channel.subscribe('connection.opened', (msg) => {
    console.log('connection opened: ', msg.data);
});
```

Reply:
Very good point. Applied the same logic to the `channel.lifecycle` example too.